lib/usdUfe/ufe/UsdUndoClearSceneItemMetadataCommand.cpp

```diff
@@ -43,31 +45,23 @@ ClearSceneItemMetadataCommand::ClearSceneItemMetadataCommand(
 
 void ClearSceneItemMetadataCommand::executeImplementation()
 {
-    if (_stage) {
-        const PXR_NS::UsdPrim prim = _stage->GetPrimAtPath(_primPath);
-        if (_group.GetString().empty()) {
-            // If this is not a grouped meta data, remove the key
-            prim.ClearCustomDataByKey(TfToken(_key));
-        } else {
+    if (!_stage)
+        return;
 
-            const PXR_NS::VtValue data = prim.GetCustomDataByKey(_group);
+    const PXR_NS::UsdPrim prim = _stage->GetPrimAtPath(_primPath);
+    if (_group.empty()) {
+        // If this is not a grouped meta data, remove the key
+        prim.ClearCustomDataByKey(TfToken(_key));
+    } else {
+        PXR_NS::TfToken fullKey(_group + std::string(":") + _key);
 
-            if (!data.IsEmpty() && data.IsHolding<PXR_NS::VtDictionary>()) {
-                // Remove the key and its value.
-                if (!_key.empty()) {
-                    PXR_NS::VtDictionary newDict = data.UncheckedGet<PXR_NS::VtDictionary>();
-                    if (newDict.find(_key) != newDict.end()) {
-                        newDict.erase(_key);
-
-                        // Set the new data.
-                        prim.SetCustomDataByKey(_group, PXR_NS::VtValue(newDict));
-                    }
-                }
-                // Remove the group.
-                else {
-                    prim.ClearCustomDataByKey(_group);
-                }
-            }
+        // When targeting the private "SessionLayerAutodesk" metadata group,
+        // we always write in the session layer.
+        if (_group == MetadataTokens->SessionLayerAutodesk) {
```

Comment:
I was thinking that when reading we would just use the group name "Autodesk". Only during writing would we use "SessionLayer-Autodesk". But yes I can see how this could be a little confusing. Why can't we intercept it when we read and remove "SessionLayer-"? 

Let's ask design what they think the group name we save in the layer should be.